src/main/java/com/fasterxml/jackson/databind/ser/SerializerCache.java

```diff
@@ -107,29 +110,41 @@ public synchronized int size() {
      */
     public JsonSerializer<Object> untypedValueSerializer(Class<?> type)
     {
-        synchronized (this) {
+        try {
```

Comment:
Surely this would never cause issues? There's no I/O operations within `Map` (`LRUMap`), `TypeKey` is stateless.

In fact... I don't think synchronization should be needed at all here: `LRUMap` states:

```
Since Jackson 2.14, the implementation
...
Is thread-safe and does NOT require external synchronization
```

Reply:
After reading through the code, this may not be as simple as I thought (see #4442): I think `synchronized` actually has secondary purpose to act as monitor for resolution of cyclic dependencies for POJO serializers (`BeanSerializer`). And if so it cannot really be safely removed (but is also a big potential performance concern at same time).